### PR TITLE
Rename FindCommand to SearchCommand

### LINKED ISSUE #24: Rename FindCommand.java to SearchCommand.java
logic --> commands --> FindCommand.java

### PR DESCRIPTION
**Pull request by johannagwan**
Resolves #24 
 
**main:**
- Rename FindCommand.java to SearchCommand.java (in src/main/java/seedu/address/logic/commands/SearchCommand.java)
- Rename FindCommandParser.java to SearchCommandParser.java (in src/main/java/seedu/address/logic/parser/SearchCommandParser.java)
- Update AddressBookParser.java accordingly (in src/main/java/seedu/address/logic/parser/AddressBookParser.java)

**test:**
- Rename FindCommandTest.java to SearchCommandTest.java (in src/test/java/seedu/address/logic/commands/SearchCommandTest.java)
- Update AddressBookParserTest.java accordingly (in src/test/java/seedu/address/logic/parser/AddressBookParserTest.java)
- Rename FindCommandParserTest.java to SearchCommandParserTest.java (in src/test/java/seedu/address/logic/parser/SearchCommandParserTest.java)

### CODE REVIEW

**Comment by kevinputera** [comment]
@johannagwan I haven't taken a look at this, will do soon.

Nonetheless, do keep a lookout at the Travis CI checks for every PR you submit. If it fails, that generally means: (1) there are compilation errors, (2) there are stylistic errors, or (3) there are test errors.

Failing checks will most likely happen quite frequently for this project since we're refactoring heavily; as such, most of the time we might need to fix the test cases to accommodate for the refactoring. However, I believe that this PR should not break, since it only involves renaming, so do fix this!

**Comment by kevinputera** [comment]
Another thing, let's add the `status.onGoing` tag to issue #24 so that we can track our progress easier in the milestone + let others know that you are working on this issue. In addition, let's [link](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) this PR to the issue, so that the issue is automatically closed when this PR is merged.

I have done the above, but writing this here for future reference

**Review by kevinputera — COMMENTED** [review]
Hey Johanna! Just some general comments:

1. Please ignore the `.vs` directory. Put the `/.vs` line in the `.gitignore` file, and comment appropriately on what these ignored files are.
2. Please remove your changes for `tutorial-add-command` from this PR! e.g. the `Remark` and `RemarkCommand` classes
3. Don't rename the URL at `README.adoc` and `build.gradle` to your forked repo!
4. I'm not sure if the inline comments you added to, for e.g.: the `LogicManager` class, helps with understanding. TBH, I feel that these additional comments make the code harder to read, as now we'll need to sieve through the lines to find the actual code. Good code is usually self-explanatory!

**Review by HemanshuGandhi — CHANGES_REQUESTED** [review]
- There are some checkstyle errors that need to be fixed in several files, mostly relating to imports. Make sure you fix them before merging. 
-  As Kevin mentioned, and according to CS2103T guidelines, inline comments are actually not recommended, possibly penalizable. I think should remove.
- As Kevin mentioned, there are commits belonging to the tutorials (RemarkCommand) that you should remove.

Otherwise LGTM!

**Comment by johannagwan** [comment]
Hi @kevinputera @HemanshuGandhi, thanks for the review! Just want to clarify, I did not add any inline comments at all. But it's a good catch! We all can take note of this in the future.

I have amended my PR as such:
- added .vs directory into gitignore
- removed unnecessary commits (I'm sorry for the mishaps)

**Review by HemanshuGandhi — APPROVED** [review]

File: src/main/java/com/notably/logic/commands/AddCommand.java
Comment (kevinputera) [inline src/main/java/com/notably/logic/commands/AddCommand.java]: You need a newline character after the end of the file. Otherwise, this will account to a style error (although we can't really check right now since the build is failing)

Comment (kevinputera) [inline src/main/java/com/notably/logic/commands/AddCommand.java:67]: You need a newline character after the end of the file. Otherwise, this will account to a style error (although we can't really check right now since the build is failing)

File: src/main/java/com/notably/logic/commands/EditCommand.java
Comment (kevinputera) [inline src/main/java/com/notably/logic/commands/EditCommand.java:26]: Please remove all the above imports. These conflicted with the removal of these imports from #48 (merged PR)

File: src/main/java/com/notably/model/Model.java
Comment (kevinputera) [inline src/main/java/com/notably/model/Model.java:7]: Please remove this import, conflict with #48

File: src/test/java/com/notably/logic/commands/DeleteCommandTest.java
Comment (kevinputera) [inline src/test/java/com/notably/logic/commands/DeleteCommandTest.java:19]: Please remove this import, conflict with #48

File: src/test/java/com/notably/logic/parser/AddCommandParserTest.java
Comment (kevinputera) [inline src/test/java/com/notably/logic/parser/AddCommandParserTest.java:135]: How do these removals relate to this PR?

File: src/test/java/com/notably/logic/parser/AddressBookParserTest.java
Comment (kevinputera) [inline src/test/java/com/notably/logic/parser/AddressBookParserTest.java:27]: Please remove this, conflict with #48

File: src/test/java/com/notably/testutil/EditPersonDescriptorBuilder.java
Comment (kevinputera) [inline src/test/java/com/notably/testutil/EditPersonDescriptorBuilder.java:12]: Please remove these imports, conflict with #48

File: src/test/java/com/notably/testutil/PersonBuilder.java
Comment (kevinputera) [inline src/test/java/com/notably/testutil/PersonBuilder.java:10]: Same here too, conflict with #48

File: src/test/java/com/notably/testutil/PersonUtil.java
Comment (kevinputera) [inline src/test/java/com/notably/testutil/PersonUtil.java:13]: Same here, conflict with #48

**Review by kevinputera — CHANGES_REQUESTED** [review]
Great job Johanna! Although @HemanshuGandhi has approved your PR, please take some time to read through some tips + considerations below. In addition, don't merge this PR until your changes from #44 are reverted from the master branch.

Consideration:
Previously, we created this issue with the intention of executing search operations through the `Parser` + `Command` flow. However, with our latest architecture, search operations will be conducted from the `SuggestionEngine` flow. I was thinking that, since this is the case, let's pivot this PR to instead remove all `FindCommand` instances, rather than renaming `FindCommand` to `SearchCommand`?

Tips:
- As a rule of thumb, please check the changes you've made after creating your PR. Sometimes, it can be hard to track what changes you've added along the way while coding. To do this, you can open the "Files changed" tab after you've created your PR, and do a self-review.
- A PR should only contain the changes needed to achieve the goal of the PR. I noticed that there are some unrelated changes in this PR, which should not be the case. For example, we can see that in this PR you're modifying the `AddCommandParserTest` class, and also made some changes that conflicted with #44. If these are changes that needed to be made to achieve your goal, please add some explanation in the description of the PR, so that your reviewer can understand your changes better.
- Now, the above two tips come hand-in-hand. By doing a self-review on the changes you've made for a PR, you can place yourself in the shoe of your reviewer and question the changes you've made. Are there any unrelated changes? Have I documented unrelated changes in the description of my PR? Did I accidentally committed something I never intended to? etc. This simple action will undoubtedly improve the quality of your PRs significantly!

**Comment by johannagwan** [comment]
I am closing this PR as we have decided to integrate the Search feature into our SuggestionEngine. I will create a new PR for the deletion of FindCommand.

**Pull request closed without merging**
